--- a/contrib/babelfishpg_tds/Makefile	2023-01-23 11:35:47.000000000 +0000
+++ b/contrib/babelfishpg_tds/Makefile	2023-01-28 20:12:41.540116219 +0000
@@ -10,6 +10,7 @@
 tds_include = $(tds_top_dir)/src/include
 TSQL_SRC = ../babelfishpg_tsql
 
+PG_CFLAGS += -Wno-error=format-security
 PG_CPPFLAGS += -I$(TSQL_SRC) -I$(PG_SRC) -I$(tds_top_dir)
 
 # Exclude the following files from the build (sometimes these
