diff --git a/contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql b/contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql
index 6679b8e2..2425cd12 100644
--- a/contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql
+++ b/contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql
@@ -511,14 +511,14 @@ BEGIN
 			out_scale as SCALE,
 			out_radix as RADIX,
 			out_nullable as NULLABLE,
-			out_remarks as REMARKS,
-			out_column_def as COLUMN_DEF,
+			cast(out_remarks as varchar(254)) as REMARKS,
+			cast(out_column_def as sys.nvarchar(4000)) as COLUMN_DEF,
 			out_sql_data_type as SQL_DATA_TYPE,
 			out_sql_datetime_sub as SQL_DATETIME_SUB,
 			out_char_octet_length as CHAR_OCTET_LENGTH,
 			out_ordinal_position as ORDINAL_POSITION,
-			out_is_nullable as IS_NULLABLE,
-			(
+			cast(out_is_nullable as varchar(254)) as IS_NULLABLE,
+			cast((
 			CASE
 				WHEN out_ss_is_identity = 1 AND out_sql_data_type = -6 THEN 48 -- Tinyint Identity
 				WHEN out_ss_is_identity = 1 AND out_sql_data_type = 5 THEN 52 -- Smallint Identity
@@ -528,7 +528,7 @@ BEGIN
 				WHEN out_ss_is_identity = 1 AND out_sql_data_type = 2 THEN 63 -- Numeric Identity
 				ELSE out_ss_data_type
 			END
-			) as SS_DATA_TYPE
+			) as sys.tinyint) as SS_DATA_TYPE
 	from sys.sp_columns_100_internal(sys.babelfish_truncate_identifier(@table_name),
 		sys.babelfish_truncate_identifier(@table_owner),
 		sys.babelfish_truncate_identifier(@table_qualifier),
@@ -559,24 +559,24 @@ BEGIN
 			out_scale as SCALE,
 			out_radix as RADIX,
 			out_nullable as NULLABLE,
-			out_remarks as REMARKS,
-			out_column_def as COLUMN_DEF,
+			cast(out_remarks as varchar(254)) as REMARKS,
+			cast(out_column_def as sys.nvarchar(4000)) as COLUMN_DEF,
 			out_sql_data_type as SQL_DATA_TYPE,
 			out_sql_datetime_sub as SQL_DATETIME_SUB,
 			out_char_octet_length as CHAR_OCTET_LENGTH,
 			out_ordinal_position as ORDINAL_POSITION,
-			out_is_nullable as IS_NULLABLE,
+			cast(out_is_nullable as varchar(254)) as IS_NULLABLE,
 			out_ss_is_sparse as SS_IS_SPARSE,
 			out_ss_is_column_set as SS_IS_COLUMN_SET,
 			out_ss_is_computed as SS_IS_COMPUTED,
 			out_ss_is_identity as SS_IS_IDENTITY,
-			out_ss_udt_catalog_name as SS_UDT_CATALOG_NAME,
-			out_ss_udt_schema_name as SS_UDT_SCHEMA_NAME,
-			out_ss_udt_assembly_type_name as SS_UDT_ASSEMBLY_TYPE_NAME,
-			out_ss_xml_schemacollection_catalog_name as SS_XML_SCHEMACOLLECTION_CATALOG_NAME,
-			out_ss_xml_schemacollection_schema_name as SS_XML_SCHEMACOLLECTION_SCHEMA_NAME,
-			out_ss_xml_schemacollection_name as SS_XML_SCHEMACOLLECTION_NAME,
-			(
+			cast(out_ss_udt_catalog_name as sys.nvarchar(128)) SS_UDT_CATALOG_NAME,
+			cast(out_ss_udt_schema_name as sys.nvarchar(128)) as SS_UDT_SCHEMA_NAME,
+			cast(out_ss_udt_assembly_type_name as sys.nvarchar(128)) as SS_UDT_ASSEMBLY_TYPE_NAME,
+			cast(out_ss_xml_schemacollection_catalog_name as sys.nvarchar(128)) as SS_XML_SCHEMACOLLECTION_CATALOG_NAME,
+			cast(out_ss_xml_schemacollection_schema_name as sys.nvarchar(128)) as SS_XML_SCHEMACOLLECTION_SCHEMA_NAME,
+			cast(out_ss_xml_schemacollection_name as sys.nvarchar(128)) as SS_XML_SCHEMACOLLECTION_NAME,
+			cast((
 			CASE
 				WHEN out_ss_is_identity = 1 AND out_sql_data_type = -6 THEN 48 -- Tinyint Identity
 				WHEN out_ss_is_identity = 1 AND out_sql_data_type = 5 THEN 52 -- Smallint Identity
@@ -586,7 +586,7 @@ BEGIN
 				WHEN out_ss_is_identity = 1 AND out_sql_data_type = 2 THEN 63 -- Numeric Identity
 				ELSE out_ss_data_type
 			END
-			) as SS_DATA_TYPE
+			) as sys.tinyint) as SS_DATA_TYPE
 	from sys.sp_columns_100_internal(sys.babelfish_truncate_identifier(@table_name),
 		sys.babelfish_truncate_identifier(@table_owner),
 		sys.babelfish_truncate_identifier(@table_qualifier),
