diff --git a/contrib/babelfishpg_common/src/encoding/encoding_utils.c b/contrib/babelfishpg_common/src/encoding/encoding_utils.c
index 3531bbd7..56ec95eb 100644
--- a/contrib/babelfishpg_common/src/encoding/encoding_utils.c
+++ b/contrib/babelfishpg_common/src/encoding/encoding_utils.c
@@ -39,7 +39,7 @@ server_to_any(const char *s, int len, int encoding, int *encodedByteLen)
 	{
 		/* No conversion is possible, but we must validate the result */
 		(void) pg_verify_mbstr(encoding, s, len, false);
-		encodedByteLen = len;
+		*encodedByteLen = len;
 		return (char *) s;
 	}
 	return (char *) do_encoding_conversion((unsigned char *) s,

