diff --git a/contrib/babelfishpg_tsql/src/tsqlIface.cpp b/contrib/babelfishpg_tsql/src/tsqlIface.cpp
index 6d1b6999..8f00938b 100644
--- a/contrib/babelfishpg_tsql/src/tsqlIface.cpp
+++ b/contrib/babelfishpg_tsql/src/tsqlIface.cpp
@@ -8,6 +8,7 @@
 #pragma GCC diagnostic ignored "-Wattributes"
 
 #include "antlr4-runtime.h" // antlr4-cpp-runtime
+#include "support/Utf8.h" // antlr4-cpp-runtime
 #include "tree/ParseTreeWalker.h" // antlr4-cpp-runtime
 #include "tree/ParseTreeProperty.h" // antlr4-cpp-runtime
 
@@ -458,7 +459,7 @@ format_errmsg(const char *fmt, const char *arg1, const char *arg2);
 
 inline std::u32string utf8_to_utf32(const char* s)
 {
-	return antlrcpp::utf8_to_utf32(s, s + strlen(s));
+	return antlrcpp::Utf8::lenientDecode(std::string_view(s, strlen(s)));
 }
 
 class MyInputStream : public ANTLRInputStream
@@ -472,7 +473,7 @@ public:
     
 		void setText(size_t pos, const char *newText)
 		{
-			UTF32String	newText32 = utf8_to_utf32(newText);
+			std::u32string	newText32 = utf8_to_utf32(newText);
 
 			_data.replace(pos, newText32.size(), newText32);
 		}
@@ -557,7 +558,7 @@ void PLtsql_expr_query_mutator::run()
 		rewritten_query += query.substr(cursor); // copy remaining expr->query
 
 	// update query string with quoted one
-	std::string new_query = antlrcpp::utf32_to_utf8(rewritten_query);
+	std::string new_query = antlrcpp::Utf8::lenientEncode(rewritten_query);
 	expr->query = pstrdup(new_query.c_str());
 }
 
